pretty/src/pane.rs

```diff
@@ -160,9 +165,32 @@ where
             }
             PaneNotation::Vert { panes, style } => {
                 let child_notes: Vec<_> = panes.iter().map(|p| &p.1).collect();
-                let child_sizes: Vec<_> = panes.iter().map(|p| p.0).collect();
-                let total_height = self.rect().height() as usize;
-                let heights: Vec<_> = divvy(total_height, &child_sizes)
+                let total_fixed: usize = panes.iter().filter_map(|p| p.0.get_fixed()).sum();
+                let total_height = self.rect().height();
+                let mut available_height = total_height.saturating_sub(total_fixed as Row);
+                let child_sizes = panes
+                    .iter()
+                    .map(|p| match p.0 {
+                        PaneSize::DynHeight => {
+                            // Convert dynamic height into a fixed height, based on the currrent document.
+                            if let PaneNotation::Content { content, .. } = &p.1 {
+                                let f = get_content.clone();
```

Comment:
Why is the `get_content` closure a `FnOnce`? Could it be a `Fn` instead, to avoid the need for it to also be `Clone`?

Reply:
Because that's the only way I could get the demo crate's implementation of `get_content` to work, because it takes ownership of some captured variables. We can see if there's a way around that, if it matters.

Reply:
whee it works